app/src/main/java/com/brandon/forzenbook/view/composables/LoginComposables.kt

```diff
@@ -53,85 +48,110 @@ fun LoginContent(
     val resources = LocalContext.current.resources
     val keyboardController = LocalSoftwareKeyboardController.current
     YellowColumn {
-        LoginTitle()
-        InputInfoTextField(
-            hint = resources.getString(R.string.loginUserNameHint),
-            onTextChange = {
-                username = it
-            },
-            characterLimit = EMAIL_CHAR_LIMIT,
-            onMaxCharacterLength = {
-                emailError = it
-            },
+        ImageTitle(
+            imageId = R.drawable.forzenlogo,
+            text = resources.getString(R.string.loginTitle),
         )
-        if (emailError) {
-            TextFieldErrorText(text = resources.getString(R.string.loginCharLengthLimit))
-        }
-        if (state.isEmailError) {
-            TextFieldErrorText(text = resources.getString(R.string.loginUserNameError))
-        }
         InputInfoTextField(
-            hint = resources.getString(R.string.logincodeHint),
-            onTextChange = {
-                code = it
-            },
-            onMaxCharacterLength = {
-                codeError = it
-            },
-            keyboardOptions = KeyboardOptions(imeAction = ImeAction.Go)
+            hint = resources.getString(R.string.loginEmailHint),
+            currentText = email,
+            onTextChange = { email = it },
+            characterLimit = EMAIL_CHAR_LIMIT,
+            onMaxCharacterLength = { emailError = it },
         )
-        if (codeError) {
-            TextFieldErrorText(text = resources.getString(R.string.loginCharLengthLimit))
+        if (emailError) TextFieldErrorText(text = resources.getString(R.string.loginCharLengthLimit))
+        if (state.isEmailError) TextFieldErrorText(text = resources.getString(R.string.loginEmailError))
+        if (state.isCodeSent) {
+            var codeError by rememberSaveable { mutableStateOf(false) }
+            InputInfoTextField(
+                hint = resources.getString(R.string.loginCodeHint),
+                onTextChange = { code = it },
+                onMaxCharacterLength = { codeError = it },
+                keyboardOptions = KeyboardOptions(imeAction = ImeAction.Go)
+            )
+            if (codeError) TextFieldErrorText(text = resources.getString(R.string.loginCharLengthLimit))
         }
-        BlackButton(text = resources.getString(R.string.loginButtonText)) {
+        BlackButton(
+            text = if (state.isCodeSent) resources.getString(R.string.loginButtonText) else resources.getString(
+                R.string.loginGetCodeText
+            )
+        ) {
             keyboardController?.hide()
-            if (username.isNotEmpty() && code.isNotEmpty()) {
-                onSubmit(username, code)
-            } else {
-                emptyFieldError = true
-            }
-        }
-        if (emptyFieldError) {
-            TextFieldErrorText(text = resources.getString(R.string.requiredFieldsError))
+            if (onCheckConnection()) {
```

Comment:
You sure you want to reinvoke this every composition? That could be a ton of unneeded work

Reply:
Seems like this should be caused by an event, rather than just happening. Ill look into this and see how to make more sense of it.

Reply:
Looking at this again, this is inside of the onclick event in the button, so it should only invoke when the button is clicked to send information. Otherwise, it acts as a lambda, unless I am misunderstanding what i've done here

Reply:
Asked about this and its good, but I did clean up the logic.